src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersister.cs

```diff
@@ -52,8 +52,7 @@ public Task Complete(IContainSagaData sagaData, SynchronizedStorageSession sessi
             VersionedSagaEntity result;
             if (data.TryGetValue(sagaId, out result) && result?.SagaData is TSagaData)
             {
-                var clone = (TSagaData) DeepClone(result.SagaData);
-                result.RecordRead(clone, version);
+                var clone = result.Read<TSagaData>(version);
```

Comment:
Since `DeepClone` is now an implementation detail of `Read`, perhaps this variable should be named something other than `clone`?